packages/jira-adapter/src/change_validators/automation/automations_issue_types.ts

```diff
@@ -0,0 +1,148 @@
+  componentElemID: ElemID
+  invalidIssueType: string | undefined
+}
+
+type ProjectWithIssueTypeScheme = InstanceElement & { value: { issueTypeScheme: ReferenceExpression } }
+
+const AUTOMATION_ISSUE_TYPE_OBJECT_SCHEME = Joi.object({
+  fieldType: Joi.string().required(),
+  value: Joi.object({
+    type: Joi.string().required(),
+    value: Joi.object().required(),
+  })
+    .unknown(true)
+    .required(),
+})
+  .unknown(true)
+  .required()
+
+const isIssueTypeObject = createSchemeGuard<IssueTypeObject>(AUTOMATION_ISSUE_TYPE_OBJECT_SCHEME)
+
+const getIssueCreateActionProject = (
+  value: Values,
+  instanceProject: ReferenceExpression | string | undefined,
+): string | undefined => {
+  if (value.value?.value === 'current') {
+    return isReferenceExpression(instanceProject) ? instanceProject.elemID.getFullName() : undefined
+  }
+  if (isReferenceExpression(value.value?.value)) {
+    return value.value.value.elemID.getFullName()
+  }
+  return undefined
+}
+
+const isInstanceWithInvalidIssueType = (
+  instance: InstanceElement,
+  projectNameToIssueTypeNames: Record<string, string[]>,
+): IssueTypeError[] => {
+  const invalidIssueTypes: IssueTypeError[] = []
+  let projectFullName: string | undefined
+
+  const instanceProject =
+    Array.isArray(instance.value.projects) && instance.value.projects.length === 1
+      ? instance.value.projects[0].projectId
+      : undefined
+
+  walkOnValue({
+    elemId: instance.elemID.createNestedID('components'),
+    value: instance.value.components,
+    func: ({ value, path }) => {
+      if (_.isPlainObject(value) && value.fieldType === PROJECT_FIELD) {
+        projectFullName = getIssueCreateActionProject(value, instanceProject)
+        return WALK_NEXT_STEP.RECURSE
+      }
+
+      if (
+        isIssueTypeObject(value) &&
+        value.fieldType === ISSUE_TYPE_FIELD &&
+        isReferenceExpression(value.value.value)
+      ) {
+        const issueTypeElemID = value.value.value.elemID
+        const isValidIssueType = projectFullName
+          ? projectNameToIssueTypeNames[projectFullName]?.includes(issueTypeElemID.getFullName())
+          : true
+        if (!isValidIssueType) {
+          invalidIssueTypes.push({ componentElemID: path, invalidIssueType: issueTypeElemID.name })
+        }
+      }
+
+      return WALK_NEXT_STEP.RECURSE
+    },
+  })
+
+  return invalidIssueTypes
+}
+
+const isIssueCreateActionAutomation = (components: Values[]): boolean =>
+  Array.isArray(components) && components.some(component => component.type === 'jira.issue.create')
+
+const isProjectInstanceWithIssueTypeScheme = (instance: InstanceElement): instance is ProjectWithIssueTypeScheme =>
+  isReferenceExpression(instance.value.issueTypeScheme)
+
+export const automationIssueTypeValidator: ChangeValidator = async (changes, elementsSource) => {
+  if (elementsSource === undefined) {
+    return []
+  }
+
+  const projects = await getInstancesFromElementSource(elementsSource, [PROJECT_TYPE])
```

Comment:
Optimization suggestion - getting elements from the element source may be expensive in large envs (or in envs with multiple projects in this case)
The current code queries the element source also when there are no relevant changes, which means that in 100% of deployments, we'll spend the time it takes to get those elements, even when it's not needed


So the suggestion is to have an early return in case there are no relevant changes in the deployment

```suggestion
    const relevantChanges =   return changes
      .filter(isInstanceChange)
      .filter(isAdditionOrModificationChange)
      .map(getChangeData)
      .filter(instance => instance.elemID.typeName === AUTOMATION_TYPE)
      .filter(instance => isIssueCreateActionAutomation(instance.value.components))
 
 if (_.isEmpty(relevantChanges)) {
    return []
 }
```